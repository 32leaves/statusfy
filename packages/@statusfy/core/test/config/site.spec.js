import path from 'path'

import generateConfig from '@/core/lib/config/generate'
import loadConfig from '@/core/lib/config/load'

const tempPath = path.resolve(__dirname, '../.tmp')

describe('config:site', () => {
  const configLabels = ['simple', 'cli', 'advanced']

  configLabels.forEach(label => {
    test(label, () => {
      const sourceDir = path.resolve(tempPath, label)
      const { siteConfig } = generateConfig(sourceDir, {})
      const siteConfigSourceDir = siteConfig.sourceDir

      siteConfig.sourceDir = path.relative(__dirname, sourceDir)

      expect(path.isAbsolute(siteConfigSourceDir)).toBeTruthy()
      expect(siteConfig).toMatchSnapshot()
    })
  })

  test('invalid', () => {
    const sourceDir = path.resolve(tempPath, 'invalid')
    const loadedConfig = loadConfig(sourceDir)

    const siteConfigErrors = loadedConfig.errors

    expect(siteConfigErrors).toMatchSnapshot()
  })
})
