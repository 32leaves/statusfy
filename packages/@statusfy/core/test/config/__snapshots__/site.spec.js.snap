// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config:site advanced 1`] = `
Object {
  "analytics": Object {
    "ga": "UA-XXXXXXXXX-Y",
  },
  "baseUrl": "https://demo.statusfy.co",
  "build": Object {
    "isStatic": false,
  },
  "content": Object {
    "dir": "content",
    "frontMatterFormat": "yaml",
    "systems": Array [
      "cdn",
      "conversions",
      "site-delivery",
      "api",
    ],
  },
  "defaultLocale": "en",
  "description": "Description",
  "head": Object {
    "link": Array [
      Object {
        "color": "#3e4e88",
        "href": "/icons/safari-pinned-tab.svg",
        "rel": "mask-icon",
      },
    ],
    "meta": Array [
      Object {
        "charset": "utf-8",
      },
      Object {
        "content": "width=device-width, initial-scale=1",
        "name": "viewport",
      },
      Object {
        "content": "Statusfy",
        "name": "generator",
      },
      Object {
        "content": "",
        "hid": "description",
        "name": "description",
      },
      Object {
        "content": "",
        "hid": "description",
        "name": "description",
      },
      Object {
        "content": "Description",
        "hid": "description",
        "name": "description",
      },
    ],
    "title": "Title",
    "titleTemplate": "%s | Title",
  },
  "locales": Array [
    Object {
      "code": "en",
      "file": "en.json",
      "iso": "en-US",
      "name": "English",
    },
    Object {
      "code": "es",
      "file": "es.json",
      "iso": "es-ES",
      "name": "Espa√±ol",
    },
  ],
  "manifest": true,
  "name": "title",
  "serviceWorker": true,
  "sourceDir": "../.tmp/advanced",
  "theme": Object {},
  "title": "Title",
}
`;

exports[`config:site cli 1`] = `
Object {
  "baseUrl": "/",
  "build": Object {
    "isStatic": false,
  },
  "content": Object {
    "dir": "content",
    "systems": Array [
      "api",
      "cdn",
      "dns",
      "site-delivery",
    ],
  },
  "defaultLocale": "en",
  "description": "",
  "locales": Array [
    Object {
      "code": "en",
      "file": "en.json",
      "iso": "en-US",
      "name": "English",
    },
  ],
  "manifest": true,
  "name": "title",
  "serviceWorker": true,
  "sourceDir": "../.tmp/cli",
  "theme": Object {},
  "title": "Title",
}
`;

exports[`config:site invalid 1`] = `
Array [
  "The default Front Matter Format ([33mcontent.frontMatterFormat[39m) is invalid. These are the valid formats: [36myaml, yml, toml, json[39m.",
  "The Base URL ([33mbaseUrl[39m) is invalid. Example: [36mhttps://status.yourbaseurl.com[39m.",
  "The Default Locale ([33mdefaultLocale[39m) value must be included in the locales list. Current value [36mit[39m, defined codes: [36men, es[39m.",
]
`;

exports[`config:site simple 1`] = `
Object {
  "baseUrl": "/",
  "build": Object {
    "isStatic": false,
  },
  "content": Object {
    "dir": "content",
    "systems": Array [
      "api",
      "cdn",
      "dns",
      "site-delivery",
    ],
  },
  "defaultLocale": "en",
  "description": "",
  "locales": Array [
    Object {
      "code": "en",
      "iso": "en-US",
      "name": "English",
    },
  ],
  "manifest": false,
  "name": "title",
  "serviceWorker": false,
  "sourceDir": "../.tmp/simple",
  "theme": Object {},
  "title": "Title",
}
`;
